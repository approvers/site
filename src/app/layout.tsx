import { Metadata, Viewport } from "next";
import { Box } from "@chakra-ui/react";
import { Footer } from "./components/footer";
import { GoogleAnalytics } from "./components/google-analytics";
import { Header } from "./components/header";
import { Navigation } from "./components/navigation";
import { Providers } from "./providers";
import { Suspense } from "react";
import { roboto } from "./fonts";

const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || "";
const siteImage = "/android-chrome-512x512.png";
const siteName = "限界開発鯖";

console.log(baseUrl);

export const viewport: Viewport = {
  themeColor: "#000000",
};

export const metadata: Metadata = {
  icons: [siteImage],
  title: {
    default: siteName,
    template: `${siteName} - %s`,
  },
  openGraph: {
    type: "website",
    title: siteName,
    siteName: siteName,
    images: [siteImage],
    locale: "ja_JP",
  },
  twitter: {
    card: "summary",
    title: siteName,
    images: [siteImage],
  },
};

const IconMetas = (): JSX.Element => (
  <>
    <link rel="apple-touch-icon" sizes="180x180" href={"/apple-touch-icon.png"} />
    <link rel="icon" type="image/png" sizes="32x32" href={"/favicon-32x32.png"} />
    <link rel="icon" type="image/png" sizes="16x16" href={"/favicon-16x16.png"} />
    <link rel="mask-icon" href={"/safari-pinned-tab.svg"} color="#000000" />
  </>
);

const headerLinks = [
  {
    name: "ホーム",
    url: "/",
  },
  {
    name: "メンバー",
    url: "/member",
  },
  {
    name: "ブログ",
    url: "/blog",
  },
  {
    name: "リンク",
    url: "/link",
  },
] as const;

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="ja" className={roboto.className}>
      <head>
        <link
          rel="stylesheet"
          href="https://cdn.jsdelivr.net/npm/yakuhanjp@3.3.1/dist/css/yakuhanjp.min.css"
          media="print"
        />
        <IconMetas />
        <link rel="manifest" href={"/site.webmanifest"} />
      </head>
      <body>
        <Suspense>
          <GoogleAnalytics />
        </Suspense>
        <Providers>
          <Navigation links={headerLinks} />
          <Header />
          <Box mt={16} mr={[8, 16]} mb={16} ml={[8, 16]}>
            {children}
          </Box>
          <Footer />
        </Providers>
      </body>
    </html>
  );
}
