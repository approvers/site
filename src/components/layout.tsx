import { Box } from "@chakra-ui/react";
import { Footer } from "./footer";
import { Header } from "./header";
import { Navigation } from "./navigation";
import { Providers } from "./providers";
import React from "react";

const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || "";
const siteImage = "/android-chrome-512x512.png";
const siteName = "限界開発鯖";

console.log(baseUrl);

export const metadata = {
  icons: [siteImage],
  title: {
    default: siteName,
    template: `${siteName} - %s`,
  },
  openGraph: {
    type: "website",
    title: siteName,
    siteName: siteName,
    images: [siteImage],
    locale: "ja_JP",
  },
  twitter: {
    card: "summary",
    title: siteName,
    images: [siteImage],
  },
};

const IconMetas = (): JSX.Element => (
  <>
    <link rel="apple-touch-icon" sizes="180x180" href={"/apple-touch-icon.png"} />
    <link rel="icon" type="image/png" sizes="32x32" href={"/favicon-32x32.png"} />
    <link rel="icon" type="image/png" sizes="16x16" href={"/favicon-16x16.png"} />
    <link rel="mask-icon" href={"/safari-pinned-tab.svg"} color="#000000" />
  </>
);

const headerLinks = [
  {
    name: "ホーム",
    url: "/",
  },
  {
    name: "メンバー",
    url: "/member",
  },
  {
    name: "ブログ",
    url: "/blog",
  },
  {
    name: "リンク",
    url: "/link",
  },
] as const;

export function Layout({ title, children }: { title: string; children: React.ReactNode }) {
  const titleText = title === "" ? siteName : `${siteName} - ${title}`;
  return (
    <>
      <html lang="ja" />
      <title>{titleText}</title>
      <link
        rel="stylesheet"
        href="https://cdn.jsdelivr.net/npm/yakuhanjp@3.3.1/dist/css/yakuhanjp.min.css"
        media="print"
      />
      <IconMetas />
      <link rel="manifest" href={"/site.webmanifest"} />
      <Providers>
        <Navigation links={headerLinks} />
        <Header />
        <Box mt={16} mr={[8, 16]} mb={16} ml={[8, 16]}>
          {children}
        </Box>
        <Footer />
      </Providers>
    </>
  );
}
