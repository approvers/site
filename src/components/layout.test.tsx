import { Layout } from "./layout";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Layout pageName="Template Layout" description="the template page for this website">
        Hello, world!
      </Layout>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
