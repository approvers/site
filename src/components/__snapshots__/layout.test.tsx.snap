// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="css-dvi6oq"
>
  <button
    className="chakra-button css-1h3syp1"
    onClick={[Function]}
    type="button"
  >
    <span
      className="chakra-button__icon css-1wh2kri"
    >
      <svg
        aria-hidden={true}
        className="svg-inline--fa fa-bars "
        data-icon="bars"
        data-prefix="fas"
        focusable={false}
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </span>
    メニュー
  </button>
  <header
    className="css-plc0gq"
  >
    <a
      className="css-70qvj9"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="chakra-avatar css-11wutbi"
      >
        <div
          aria-label="Approversロゴ"
          className="chakra-avatar__initials css-0"
          role="img"
        >
          A
        </div>
      </span>
      <span
        className="chakra-heading css-0"
      >
        Approvers
      </span>
    </a>
  </header>
  <div
    className="css-oogwee"
  >
    Hello, world!
  </div>
  <footer
    className="css-114d7fc"
  >
    <a
      className="chakra-link css-i6sr0x"
      href="/policy"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      プライバシーポリシー
    </a>
    - (c) 
    2022
     Approvers
  </footer>
</div>
`;
